### PR TITLE
refactor(multi-env): move appPackage to templates folder

### PR DESCRIPTION
**Pull request by a1exwang**
[feature design](https://microsoftapc.sharepoint.com/teams/DevDivTeamsDevXProductTeam/_layouts/OneNote.aspx?id=%2Fteams%2FDevDivTeamsDevXProductTeam%2FShared%20Documents%2FService%20infrastructure%20Team%2FInfra%20team&wd=target%28Infra.one%7C54126BB5-B660-493E-ABC2-5CDF1390B13A%2FFolder%20structure%20and%20file%20naming%7C137CF56A-5AD2-4117-9AB0-E2C7609DA551%2F%29)

- move `appPackage/appPackage.zip` to `appPackage/appPackage.{envName}.zip`
- move `appPackage/manifest.source.json` to `templates/appPackage/manifest.template.json`
- move `appPackage/*.png` to `templates/appPackage/resources/*.png`

image

https://msazure.visualstudio.com/Microsoft%20Teams%20Extensibility/_workitems/edit/10667418/

### CODE REVIEW

File: packages/fx-core/src/common/tools.ts [hunk 1]
```diff
@@ -433,13 +433,23 @@ export function compileHandlebarsTemplateString(templateString: string, context:
 
 export async function getAppDirectory(projectRoot: string): Promise<string> {
```
Comment (jayzhang) [inline packages/fx-core/src/common/tools.ts:434]: please make sure all modules referring to `appPackage` folder are adjusted accordingly.

Comment (jayzhang) [reply packages/fx-core/src/common/tools.ts:434]: Including sample app

Comment (a1exwang) [reply packages/fx-core/src/common/tools.ts:434]: - I have search "appPackage" globally.
- This change will not take effect until we open the `TEAMSFX_MULTI_ENV` feature flag. We will update the sample app before opening this feature flag

**Review by jayzhang — APPROVED** [review]

File: packages/fx-core/src/common/tools.ts [hunk 2]
```diff
@@ -433,13 +433,23 @@ export function compileHandlebarsTemplateString(templateString: string, context:
...
   const appDirNewLoc = `${projectRoot}/${AppPackageFolderName}`;
   const appDirOldLoc = `${projectRoot}/.${ConfigFolderName}`;
 
-  if (await fs.pathExists(`${appDirNewLoc}/${REMOTE_MANIFEST}`)) {
-    return appDirNewLoc;
+  if (isMultiEnvEnabled()) {
+    if (await fs.pathExists(`${appDirNewLocForMultiEnv}/${MANIFEST_TEMPLATE}`)) {
+      return appDirNewLocForMultiEnv;
+    } else {
+      return appDirOldLoc;
```
Comment (dooriya) [inline packages/fx-core/src/common/tools.ts:445]: There're two possible locations for the existing manifest template locations:
1) ./fx/manifest.source.json
2) appPackage/manifest.source.json

We should cover both cases.

Comment (a1exwang) [reply packages/fx-core/src/common/tools.ts:447]: The first case is handled and added code to handle case 2

**Review by dooriya — APPROVED** [review]

**Review by nliu-ms — APPROVED** [review]
LGTM, the project migration will be done before feature flag set to true.

**Pull request merged**